### ISSUE TITLE
Font Ligature Setting does not work as intended / disable Font Ligatures

### ISSUE THREAD

**Issue by bhenrich** [zed-industries/zed#4313]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Opening zed's settings and disabling font ligatures does not in fact disable font ligatures. This is the configuration entered in the settings.json:

"buffer_font_features": {
    "liga": false
  }

Yet as shown in this screenshot the ligature remain, even after re-launching Zed. (The Font used in this screenshot is JetBrains Mono, but this also happens with other fonts supporting Ligatures.)
![Screenshot 2023-12-06 at 11 05 09](https://github.com/zed-industries/community/assets/96016408/a51ad65b-cf9d-4290-812b-aebbdab8af0b)

### Environment

Zed: v0.114.4 (stable)
OS: macOS 14.1.2
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by pepehandsjpg** [comment]
Same issue with Monolisa font

**Comment by buzbohdan** [comment]
Can you try disabling ligatures with
```json
"buffer_font_features": {
    "calt": false
}
```
AND restarting Zed? Restarting Zed worked for me with the default font. I can't check `Monolisa` or `Cartograph CF` as they are not easy to obtain

**Comment by estebangarcia21** [comment]
Same issue here. Attempting to enable / disable certain OpenType features for Jetbrains Mono (and Fira Code) has no effect. Im using Zed v0.123.3
```json
{
  "buffer_font_family": "JetBrains Mono",
  "buffer_font_features": {
    "ss02": true,
    "zero": true,
    "calt": false
  }
}
```

**Comment by amio** [comment]
Take "Fira Code" as an example, the lingatures listed in their [doc](https://github.com/tonsky/FiraCode):
![image](https://github.com/zed-industries/zed/assets/215282/80b60ba1-c084-4661-8412-e507ff7fbb8d)

this configuration:
```
  "buffer_font_family": "Fira Code",
  "buffer_font_features": {
    "ss02": false,
    "ss08": false
  },
```

has no effect:
![image](https://github.com/zed-industries/zed/assets/215282/595c5764-7b8e-4fe7-ae29-fe64b305db8d)

**Comment by Yesterday17** [comment]
It seems that live reloading of `buffer_font_features` does not work. `"calt": false` worked for me when using `JetBrains Mono` after restarting zed.

**Comment by JosephTLyons** [comment]
This is now available in [Zed Preview 0.127.0](https://zed.dev/releases/preview/0.127.0).

**Comment by Jeremi360** [comment]
I using Zed 0.171.6 and try to use ligatures with font "Monaspace Radon",
but restarting Zed, rechoosing this font or adding `font_fetures` doesn't seem to work :/
